client/src/pages/recording-view/components/annotation/annotation-viewer.tsx

```diff
@@ -24,6 +77,11 @@ const AnnotationViewer = ({ currentFFT }: AnnotationViewerProps) => {
     setSelectedAnnotation,
   } = useSpectrogramContext();
   const lower_freq = meta.getCenterFrequency() - meta.getSampleRate() / 2;
+  const [editAnnotationLabelY, setEditAnnotationLabelY] = useState(null);
+  const [editAnnotationLabelX, setEditAnnotationLabelX] = useState(null);
+  const [editAnnotationLabelId, setEditAnnotationLabelId] = useState(null);
+  const [editAnnotationLabelText, setEditAnnotationLabelText] = useState(null);
+  const [editAnnotationLabelChanged, setEditAnnotationLabelChanged] = useState(false);
 
```

Comment:
would it be easier to do@

const annotationTextarea = useRef(null);
{editAnnotationLabelY && editAnnotationLabelId !== null && (< textarea ref={annotationTextarea} /> )} ?

Reply:
maybe, but I would need more detail to see what you mean.